src/components/MarketingConsent/MarketingConsentCheckbox.js

```diff
@@ -199,25 +201,42 @@ class MarketingConsentCheckbox extends Component {
           {
             item.field.map(field => (
               <div key={field.id} className="form__field--wrapper form__field--sub-field-wrapper">
-                <InputField
-                  type={field.type}
-                  id={field.name}
-                  name={field.name}
-                  required={field.required}
-                  placeholder={field.placeholder}
-                  label={field.label}
-                  pattern={field.pattern}
-                  helpText={field.helpText}
-                  isValid={(valid, name) => {
-                    this.setInputValidity(name, valid, checkbox);
-                  }}
-                  emptyFieldErrorText={field.errorMessage}
-                  showErrorMessage={this.props.showErrorMessages}
-                  fieldValue={this.state.checkboxValidation[checkbox].fieldValidation[field.name]}
-                  value={() => this.fieldValue(checkbox, field.name)}
-                  // Only use our improved Yup-based validation for the problematic fields
-                  yupValidation={field.type === 'email' || field.type === 'tel'}
-                />
+
+                {field.id === 'country' ?
+
+                  <SelectField
+                    id="country"
```

Comment:
Just as a point, it'd be nice to see this use the object properties as per the Input here; 'one source of truth' and all that

Reply:
For sure. This isn't the final version.